core/src/net/sf/openrocket/aerodynamics/BarrowmanCalculator.java

```diff
@@ -515,39 +515,46 @@ private double calculatePressureDrag(FlightConfiguration configuration, FlightCo
 		base = calculateBaseCD(conditions.getMach());
 		
 		total = 0;
-		for (RocketComponent c : configuration.getActiveComponents()) {
+		final InstanceMap imap = configuration.getActiveInstances();
+	    for(Map.Entry<RocketComponent, ArrayList<InstanceContext>> entry: imap.entrySet() ) {
+			final RocketComponent c = entry.getKey();
 			if (!c.isAerodynamic())
 				continue;
-			
-			// Pressure fore drag
-			double cd = calcMap.get(c).calculatePressureDragForce(conditions, stagnation, base,
-					warnings);
-			total += cd;
-			
-			if (map != null) {
-				map.get(c).setPressureCD(cd);
-			}
-			
-			if(c.isCDOverridden()) continue;					
 
-	
-			// Stagnation drag
-			if (c instanceof SymmetricComponent) {
-				SymmetricComponent s = (SymmetricComponent) c;
+			// iterate across component instances
+			final ArrayList<InstanceContext> contextList = entry.getValue();
+			for(InstanceContext context: contextList ) {
 				
-				if (radius < s.getForeRadius()) {
-					double area = Math.PI * (pow2(s.getForeRadius()) - pow2(radius));
-					cd = stagnation * area / conditions.getRefArea();
-					total += cd;
-					if (map != null) {
-						map.get(c).setPressureCD(map.get(c).getPressureCD() + cd);
-					}
+				// Pressure fore drag
+				double cd = calcMap.get(c).calculatePressureDragForce(conditions, stagnation, base,
+																	  warnings);
+				total += cd;
+				
+				if (map != null) {
+					map.get(c).setPressureCD(cd);
```

Comment:
nit: I realize this is carry-over, however there's a couple of places in this path which lookup the AerodynamicForces for the component and use it. Each time I had to re-lookup what the map.get(c) was returning. I wonder if it would be more clear if the AerodynamicForces was set at the beginning of the loop and use it throughout. i.e.

```java
AerodynamicForces forces = (map != null) ? map.get(c) : null;
...
if (forces != null) {
    forces.setPressureCD(cd);
}
```

Reply:
I noticed a few possible optimizations like that while I was working on it -- I was thinking for this PR I wanted to stick with just fixing things that are truly broken, and then maybe revisit later.  That would still be my preference, but if others would like to see things like this improved while I'm at it I certainly can.

Reply:
Works for me. Additional changes are perfectly fine outside of this PR.